File: app/src/main/java/online/fatbook/fatbookapp/ui/activity/fill_additional_info/FillAdditionalInfoViewModel.java
package online.fatbook.fatbookapp.ui.activity.fill_additional_info;

import android.app.Application;
import android.content.Intent;
import android.view.View;
import android.widget.Toast;

import androidx.annotation.NonNull;
import androidx.lifecycle.AndroidViewModel;
import androidx.lifecycle.LiveData;
import androidx.lifecycle.MutableLiveData;

import com.fatbook.fatbookapp.R;

import java.io.File;
import java.util.Map;

import online.fatbook.fatbookapp.core.User;
import online.fatbook.fatbookapp.ui.activity.MainActivity;
import online.fatbook.fatbookapp.util.UserUtils;

public class FillAdditionalInfoViewModel extends AndroidViewModel {

    private final MutableLiveData<Map<String, Object>> mMap;

    public FillAdditionalInfoViewModel(@NonNull Application application) {
        super(application);
        mMap = new MutableLiveData<>();
        fillData(mMap);
    }

    private void fillData(MutableLiveData<Map<String, Object>> mMap) {
        //TODO fill data for test
    }

    public LiveData<Map<String, Object>> getMap() {
        return mMap;
    }

    public void saveUser(View view, User user, File image) {
        Intent intent = new Intent(getApplication(), MainActivity.class);
        if (UserUtils.createNewUser(user, image)) {
            intent.putExtra(UserUtils.TAG_USER, user);
            view.getContext().startActivity(intent);
        } else {
            Toast.makeText(getApplication(), R.string.ERROR_CREATING_USER, Toast.LENGTH_SHORT).show();
        }
    }
}

File: app/src/main/java/online/fatbook/fatbookapp/ui/activity/fill_additional_info/FillAdditionalInfoViewModel.kt
package online.fatbook.fatbookapp.ui.activity.fill_additional_info

import android.app.Application
import android.content.Intent
import android.view.View
import androidx.lifecycle.AndroidViewModel
import androidx.lifecycle.LiveData
import androidx.lifecycle.MutableLiveData
import online.fatbook.fatbookapp.core.User
import online.fatbook.fatbookapp.ui.activity.MainActivity
import java.io.File

class FillAdditionalInfoViewModel(application: Application) : AndroidViewModel(application) {

    private val mMap: MutableLiveData<Map<String, Any>> = MutableLiveData()

    private fun fillData(mMap: MutableLiveData<Map<String, Any>>) {
        //TODO fill data for test
    }

    val map: LiveData<Map<String, Any>>
        get() = mMap

    fun saveUser(view: View?, user: User?, image: File?) {
        val intent = Intent(getApplication(), MainActivity::class.java)
        //        if (UserUtils.createNewUser(user, image)) {
//            intent.putExtra(UserUtils.TAG_USER, user);
//            view.getContext().startActivity(intent);
//        } else {
//            Toast.makeText(getApplication(), R.string.ERROR_CREATING_USER, Toast.LENGTH_SHORT).show();
//        }
    }

    init {
        fillData(mMap)
    }
}